### PR TITLE
Ability to set guild_ids for all commands in a cog

### PR DESCRIPTION
**Pull request by Middledot**
## Summary
```python
class SomeCog(commands.Cog, guild_ids=[...]):
    def __init__(self, client):
        self.client = client

    @discord.slash_command(name="locale")
    async def n(self, ctx):
        await ctx.respond(ctx.interaction.locale)

    @discord.slash_command(name="guild-locale")
    async def h(self, ctx):
        await ctx.respond(ctx.interaction.guild_locale)
```
Would be a convenient alternative to putting `guild_ids` in each command or using `command_attrs`

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Review by Dorukyum — APPROVED** [review]
Looks good to me

File: discord/cog.py [hunk 1]
```diff
@@ -213,6 +220,8 @@ def __new__(cls: Type[CogMeta], *args: Any, **kwargs: Any) -> CogMeta:
 
         # Update the Command instances dynamically as well
         for command in new_cls.__cog_commands__:
+            if isinstance(command, ApplicationCommand) and command.guild_ids is None and len(new_cls.guild_ids) != 0:
+                command.guild_ids = new_cls.guild_ids
```
Comment (ooliver1) [inline discord/cog.py:224]: shouldnt this be `__cog_guild_ids__`

File: discord/cog.py [hunk 2]
```diff
@@ -137,6 +143,7 @@ def __new__(cls: Type[CogMeta], *args: Any, **kwargs: Any) -> CogMeta:
         name, bases, attrs = args
         attrs["__cog_name__"] = kwargs.pop("name", name)
         attrs["__cog_settings__"] = kwargs.pop("command_attrs", {})
+        attrs["__cog_guild_ids__"] = kwargs.pop("guild_ids", [])
```
Comment (ooliver1) [inline discord/cog.py:146]: shouldnt this be added to the `TYPE_CHECKING` guarded annotations to this

**Review by ooliver1 — CHANGES_REQUESTED** [review]
(note not just for you, but all py-cord managers)
please test prs - this pr does not use the dunder `__cog_guild_ids__` which you would notice if you did test this

File: discord/cog.py (hunk 1 above)
Comment (Dorukyum) [reply discord/cog.py:224]: Have you checked the definition of `new_cls`

Comment (ooliver1) [reply discord/cog.py:224]: `new_cls = super().__new__(cls, name, bases, attrs, **kwargs)` - everywhere else uses the dunder attributes

Comment (ooliver1) [reply discord/cog.py:224]: image of py-cord file - ignore all the other errors as i just copied raw text but heres a screenshot

Comment (Middledot) [reply discord/cog.py]: I tested before changing and missed that, thanks

**Review by Lulalaby — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]

**Pull request merged**